pkg/component/controller/controllersleasecounter.go

```diff
@@ -102,3 +110,52 @@ func (l *K0sControllersLeaseCounter) Stop() error {
 	}
 	return nil
 }
+
+// Check the numbers of controller every 10 secs and notify the subscribers
+func (l *K0sControllersLeaseCounter) runLeaseCounter(ctx context.Context) {
+	log := logrus.WithFields(logrus.Fields{"component": "controllerlease"})
+	log.Debug("starting controller lease counter every 10 secs")
+	ticker := time.NewTicker(10 * time.Second)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			log.Info("stopping controller lease counter")
+			return
+		case <-ticker.C:
+			log.Debug("counting controller lease holders")
+			count, err := l.countLeaseHolders(ctx)
+			if err != nil {
+				log.Errorf("failed to count controller leases: %s", err)
+			}
+			l.notifySubscribers(count)
+		}
+	}
+}
+
+func (l *K0sControllersLeaseCounter) countLeaseHolders(ctx context.Context) (int, error) {
+	client, err := l.KubeClientFactory.GetClient()
+	if err != nil {
+		return 0, err
+	}
+
+	return kubeutil.GetControlPlaneNodeCount(ctx, client)
+}
+
+// Notify the subscribers about the current controller count
+func (l *K0sControllersLeaseCounter) notifySubscribers(count int) {
+	logrus.WithFields(logrus.Fields{"component": "controllerlease"}).Debugf("notifying subscribers (%d) about controller count: %d", len(l.subscribers), count)
+	for _, ch := range l.subscribers {
+		// Use non-blocking send to avoid blocking the loop
+		select {
+		case ch <- count:
+		default:
+		}
```

Comment:
I'm not sure if this is a problem, but this will cause the subscribers to not get the update if they are not receiving messages from the channel at that exact moment or if the channel is full for example because it's still processing the previous update.

Maybe a timeout, something like:

```suggestion
		select {
		case ch <- count:
		case <-time.After(time.Second):
		}
```